packages/ipfs-http-client/src/lib/ipld-formats.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+const raw = require('ipld-raw')
+const multicodec = require('multicodec')
+
+/**
+ * @typedef {import('cids')} CID
+ */
+
+/**
+ * Return an object containing supported IPLD Formats
+ *
+ * @param {object} [options] - Options passed to the http client constructor
+ * @param {object} [options.ipld] - IPLD options passed to the http client constructor
+ * @param {Array} [options.ipld.formats] - A list of IPLD Formats to use
+ * @param {Function} [options.ipld.loadFormat] - An async function that can load a format when passed a codec name
+ * @returns {Function}
+ */
+module.exports = (options) => {
+  const formats = {
+    [multicodec.DAG_PB]: dagPB,
+    [multicodec.DAG_CBOR]: dagCBOR,
+    [multicodec.RAW]: raw
+  }
+
+  const ipldOptions = (options && options.ipld) || {}
+  const configuredFormats = (ipldOptions && ipldOptions.formats) || []
```

Comment:
```suggestion
  const configuredFormats = ipldOptions.formats || []
```
We do not need the extra validation per fallback above